Do not merge

B00M3000 opened the pull request.

B00M3000 committed:
Author Card

B00M3000 committed:
Gated create

B00M3000 committed:
icons

B00M3000 committed:
Semi-finished user pages

B00M3000 committed:
d

B00M3000 committed:
Merge branch 'master' into schematic-ownership

B00M3000 committed:
asdf

B00M3000 committed:
as

B00M3000 committed:
user replaces website name when logged in

B00M3000 committed:
f

B00M3000 committed:
s

B00M3000 committed:
d

B00M3000 committed:
corect status code

B00M3000 committed:
direct delete and edit for owners of schematic

B00M3000 committed:
d

B00M3000 committed:
Merge pull request #159 from B00M3000/schematic-ownership

Schematic Ownership

B00M3000 committed:
Added link to old database

B00M3000 committed:
Merge branch 'master' of https://github.com/B00M3000/Mindustry-Schematics

B00M3000 committed:
Fixed nothing displaying when user id is defined but not a database key

B00M3000 committed:
Added ads.txt

B00M3000 committed:
Added margin around create schematic policy text

B00M3000 committed:
Fixed Author card location and formatting on edit schematic page

B00M3000 committed:
Added colon after views on schematic info page

B00M3000 committed:
Fixed formatting on mod schematic accept and decline page

B00M3000 committed:
Merge pull request #161 from B00M3000/fix-css

Update CSS

B00M3000 committed:
Fixed user profile not circled.

B00M3000 committed:
Merge branch 'fix-css'

B00M3000 committed:
d

B00M3000 committed:
d

B00M3000 committed:
phonized

B00M3000 committed:
formatted css

B00M3000 committed:
fix

B00M3000 committed:
Nav changes

B00M3000 committed:
Merge pull request #162 from B00M3000/user-profile

User profile

JeanJPNM committed:
Refactoring and optimizations (#163)

* format project files

* fix schema typings

The old way of extending the Document interface is deprecated

* fix type errors on endpoints

* replace 308 status code by 307 on auth redirect

* remove unused submit button

* refactor user avatar component

* make mongoose work with hot reloads

* remove console.log from user store

* remove unused css from index page

* remove unusde "mode" field from SchematicQueryJSON

B00M3000 committed:
d

JeanJPNM committed:
Add v7 items and liquids (#164)

* add icons for the new items and liquids

* optimize item svgs

* optimize liquid svgs

* extract items into a reusable array

* sort the item array

JeanJPNM commented:
I converted this into a draft to make the current status of the changes clear

B00M3000 committed:
only relative

B00M3000 committed:
nothing to see here

B00M3000 committed:
prevent non hex string session_id from causing errors

B00M3000 committed:
Merge branch 'master' into login-redirect

B00M3000 committed:
d

B00M3000 committed:
Merge pull request #165 from B00M3000/login-redirect

Login redirect

B00M3000 committed:
Fixed user schematics page not re-rendering after when using in page router

JeanJPNM committed:
update general dependencies

JeanJPNM committed:
update the schematic parser

B00M3000 committed:
/schematics/parse.json endpoint returns more details on parsed schematics

B00M3000 committed:
Merge branch 'master' of https://github.com/B00M3000/Mindustry-Schematics

B00M3000 committed:
Update tsconfig.json

B00M3000 committed:
d

B00M3000 committed:
d

B00M3000 committed:
d

B00M3000 committed:
d

B00M3000 committed:
d

JeanJPNM commented on package.json:
```diff
@@ -3,10 +3,9 @@
+    "build": "vite build",
     "package": "svelte-kit package",
     "preview": "vite preview",
-    "prepare": "svelte-kit sync",
```
Revert these changes, that should make the code work again.

B00M3000 committed:
ddd

B00M3000 committed:
dd

JeanJPNM committed:
update to version 4.2.0 of the schematic parser

JeanJPNM committed:
Update all dependencies (#166)

* update project base template

* migrate routes

* update how the user store is synced

JeanJPNM committed:
fix dowloads page config

JeanJPNM committed:
update the parser

JeanJPNM committed:
actually update the parser

JeanJPNM committed:
fix user login redirect urls

B00M3000 committed:
cleaning

B00M3000 committed:
Merge branch 'master' of https://github.com/B00M3000/Mindustry-Schematics

B00M3000 committed:
Added Google Analytics Tag

JeanJPNM committed:
remove duplicated 'Impact Reactor' tag

JeanJPNM committed:
update sveltekit to version 1.0.1

JeanJPNM committed:
update dependencies

JeanJPNM committed:
fix the schematic download route

JeanJPNM committed:
make the whole schematic card clickable

B00M3000 committed:
Updated .gitignore

Ignored package-lock.json

B00M3000 committed:
Removed getenv.js

Removed Unused File

B00M3000 committed:
Added Votes Field to SchematicSchema

B00M3000 committed:
Installed Svelte Icons Pack

B00M3000 committed:
Upvote and Downvote for Show on SchematicCard

Non-Functional

B00M3000 committed:
Added SVG Icon

B00M3000 committed:
Updated BasicSchematicJSON to include votes

B00M3000 committed:
Fixed Votes Type for Schematic Schema

B00M3000 committed:
custom svgs

B00M3000 committed:
addded toast and coloring of icons

B00M3000 committed:
Moving Vote out of SchematicCard

B00M3000 committed:
Updated Gitignore

B00M3000 committed:
Removed useless getenv.js

B00M3000 committed:
Update package.json

B00M3000 committed:
Reworked MongoDB Environment Variables

Removed username, password, database, and path configuration in light of just using a preformatted uri environment variable.

B00M3000 committed:
Refactoring of fetching discord token

JeanJPNM committed:
Wider use of pagination (#169)

* create a Paginator component

* set the bottom bar height without js

* add a back button to the user profile page

* add function to calculate pagination values

* add support for pagination in the user profile page

* add a generic PaginationText component

* fix small counting error in the PaginationText component

* add pagination to the schamtic changes page

JeanJPNM committed:
load the whole background before the transition (#168)

This makes background transitions smoother on lower internet
speeds, although it also makes them lag behind the actual
navigation event, this problem can be amortized by caching the images

B00M3000 committed:
PORT Environment Variable is actually functional

B00M3000 committed:
Merge pull request #170 from B00M3000/better-env

Reworking of Environment Variables and other small changes

B00M3000 committed:
Merge branch 'master' into udimp

B00M3000 committed:
Moved Converted SVG's

I've already converted these two SVG's to my custom format, just keeping them in a separate folder in case I need to reference them. They are NOT used.

B00M3000 committed:
Added vote processing endpoint

B00M3000 committed:
Fixed incorrect usage of $set and $unset query operators

B00M3000 committed:
Include votes in schematic data passed to client

B00M3000 committed:
pass entire schematic instead of just vote count

B00M3000 committed:
continual implementation of voting

B00M3000 committed:
Converted put-litter-in-its-place into my custom icon format

B00M3000 committed:
stuff

B00M3000 committed:
Currently useless endpoint

JeanJPNM committed:
Fix schematic details page (#171)

* add a default bottom bar height

* prevent the tags div from breaking the layout

JeanJPNM committed:
formatted files

B00M3000 committed:
Rather than defaulting to alt text if the avatar url is unavaliable, it defaults to the default avatar

B00M3000 committed:
Fixed issue where if there is no schematic.tags it errors

B00M3000 committed:
added missing "!"

B00M3000 committed:
Added User Search Functionality

B00M3000 committed:
Added link to access user search functionality in the user menu

B00M3000 committed:
Removed ability to remvoe admin from control panel, only able to be done through mongoDB database editing

B00M3000 committed:
Added /users route under User nav bar section

B00M3000 committed:
Adding space underneath main body so that the voting buttons arent covered by the bottom menu

a

B00M3000 committed:
placement of voting buttons

B00M3000 committed:
Recoloring of schematic vote page div

B00M3000 committed:
Creates a new vote array if perviously undefined

B00M3000 committed:
Removed redundant file

B00M3000 committed:
Merge pull request #167 from B00M3000/udimp

Upvoting and Downvoting Implementation

B00M3000 committed:
Only one option avaliable, direct or request, not both

B00M3000 committed:
Merge pull request #173 from B00M3000/only-direct-actions-for-own-schematics

Either Direct or Request form, not both for changes and deletions

JeanJPNM committed:
Update dependencies (#174)

* update package-lock.yaml

* upgrade to typescript 5.1.3

* update prettier packages

* update eslint packages

* update dotenv

* update vite

* update playwright

* update @types/diff

* update @types/showdown

* update svelte packages

* update @zerodevx/svelte-toast

B00M3000 committed:
Updated the tags.ts and the database respectively

B00M3000 committed:
Merge pull request #175 from B00M3000/tag-fixes

Updated the tags.ts and the database respectively

B00M3000 committed:
added access selector to user search function

B00M3000 committed:
Merge pull request #172 from B00M3000/advanced-searching-and-display

More Advanced Search Functions and Interface

B00M3000 committed:
updated dependancies

B00M3000 committed:
Merge branch 'master' of https://github.com/B00M3000/Mindustry-Schematics

B00M3000 committed:
When clicking a tag in the schematic's info area it will redirect you to a search query for that tag

JeanJPNM committed:
Fix pagination bar links (#176)

* fix the links of pagination bars

the links are now invalidated every time the page's href changes

* refactor link handling code in the main page

B00M3000 committed:
Merge pull request #177 from B00M3000/8-8-23-changes

When clicking a tag in the schematic's info area it will redirect you…

B00M3000 committed:
Added creator to webhooks

B00M3000 committed:
Merge pull request #179 from B00M3000/improve-webhooks

Added creator to webhooks

JeanJPNM committed:
update the schematic parser (#180)

B00M3000 committed:
added dockerfile

B00M3000 committed:
Create docker-publish.yml

B00M3000 committed:
changed node versions

B00M3000 committed:
updating pkgs

B00M3000 committed:
no longer using replit

Signed-off-by: B00M3000 <40374754+B00M3000@users.noreply.github.com>

B00M3000 committed:
prerendering was causing issues, fix later

Signed-off-by: B00M3000 <40374754+B00M3000@users.noreply.github.com>

B00M3000 committed:
typing

Signed-off-by: B00M3000 <40374754+B00M3000@users.noreply.github.com>

B00M3000 committed:
shuffling css around

Signed-off-by: B00M3000 <40374754+B00M3000@users.noreply.github.com>

B00M3000 committed:
removed empty file

Signed-off-by: B00M3000 <40374754+B00M3000@users.noreply.github.com>

B00M3000 committed:
other changes

Signed-off-by: B00M3000 <40374754+B00M3000@users.noreply.github.com>

B00M3000 committed:
cleanup

Signed-off-by: B00M3000 <40374754+B00M3000@users.noreply.github.com>

B00M3000 committed:
up deps

Signed-off-by: B00M3000 <40374754+B00M3000@users.noreply.github.com>

B00M3000 committed:
smallchanges

Signed-off-by: B00M3000 <40374754+B00M3000@users.noreply.github.com>

B00M3000 committed:
fixed schematic upload form not working

Signed-off-by: B00M3000 <40374754+B00M3000@users.noreply.github.com>

B00M3000 committed:
added cred

Signed-off-by: B00M3000 <40374754+B00M3000@users.noreply.github.com>

B00M3000 committed:
fixed spacing

Signed-off-by: B00M3000 <40374754+B00M3000@users.noreply.github.com>
